dataclassy/dataclass.py

```diff
@@ -64,6 +60,7 @@ def __new__(mcs, name, bases, dict_, **kwargs):
         # delete what will become stale references so that Python creates new ones
         dict_.pop('__dict__', None)
         dict_.pop('__weakref__', None)
+        dict_.pop('__classcell__', None)
```

Comment:
I actually tried this and it doesn't work either.  The issue to my limited understanding is that whatever `__classcell__` is in `dict_` has to be passed to `super().__new__`.

The implementation that's currently in master gives issues when `__classcell__` is in a parent class but not in `dict_`, because then you're giving `super().__new__` a `__classcell__` when it shouldn't be getting one.  This fixes that case, but gives issues when `__classcell__` was in `dict_` and should have been given to `super().__new__`.

Reply:
I'll try to set up some tests that fail for each permutation.  No promises though, I'm out of my depth here.

Reply:
Good catch, I wasn't sure about this since I noticed removing the line completely doesn't fail any tests so adding some would be great.

Reply:
I added another test (hopefully well-commented) that fails in different ways for both `__classcell__` problems.  It also fails in the current setup for reasons I describe in the comment there.

Reply:
Hmm, OK - because this might take a little while to figure out I would quite like to merge this PR and fix `__classcell__` later so that it doesn't hold up the rest of the fixes. Maybe it is as simple as only putting attributes in defaults if they are in annotations as you suggested. I can try that out, otherwise, feel free to open another PR.